Core/NsvSlicedBlock.cs

```diff
@@ -0,0 +1,384 @@
+using SliceVisualizer.Configuration;
+using SliceVisualizer.Models;
+using UnityEngine;
+using UnityEngine.UI;
+using Zenject;
+
+namespace SliceVisualizer.Core
+{
+    internal class NsvSlicedBlock : MonoBehaviour
+    {
+        private PluginConfig _config = null!;
+        private NsvAssetLoader _assetLoader = null!;
+        private ColorManager _colorManager = null!;
+
+        private Transform _blockTransform = null!;
+        private Transform _sliceTransform = null!;
+        private Transform _missedAreaTransform = null!;
+        private Transform _sliceGroupTransform = null!;
+        private SpriteRenderer _background = null!;
+        private SpriteRenderer _arrow = null!;
+        private SpriteRenderer _circle = null!;
+        private SpriteRenderer _missedArea = null!;
+        private SpriteRenderer _slice = null!;
+
+        private float _aliveTime;
+        private bool _isDirectional;
+        private Color _color;
+        private Color _otherColor;
+        private Color _arrowColor;
+        private Color _missedAreaColor;
+        private Color _sliceColor;
+        private bool _needsUpdate;
+
+        [Inject]
+        internal void Construct(PluginConfig config, NsvAssetLoader assetLoader, ColorManager colorManager)
+        {
+            _colorManager = colorManager;
+            _assetLoader = assetLoader;
```

Comment:
`_colorManager` and `_assetLoader` are only required during build process, so no need to hold reference to those after constructing. So I suggest either not storing them or `null`ing them

Reply:
Will pass the assetLoader reference, as a parameter instead. However, the _colorManager is used multiple times later on, so I'll keep that one as is.